dialogflow/stopwatch/Stopwatch/StopwatchService.swift

```diff
@@ -38,121 +36,89 @@ enum StopwatchServiceError: Error {
+        inputAudioConfig.languageCode = "en-US"
+        inputAudioConfig.sampleRateHertz = Int32(self.sampleRate)
+        queryInput.audioConfig = inputAudioConfig
+
+        let streamingDetectIntentRequest = DFStreamingDetectIntentRequest()
+        streamingDetectIntentRequest.session = "projects/" + ProjectName +
+          "/agent/sessions/" + SessionID
+        streamingDetectIntentRequest.singleUtterance = true
+        streamingDetectIntentRequest.queryParams = self.getQueryParmasFor()
+        streamingDetectIntentRequest.queryInput = queryInput
+        streamingDetectIntentRequest.outputAudioConfig = self.getOutputAudioConfig()
+        self.writer.writeValue(streamingDetectIntentRequest)
+      }
 
+      // send a request message containing the audio data
       let streamingDetectIntentRequest = DFStreamingDetectIntentRequest()
-      streamingDetectIntentRequest.session = "projects/" + ProjectName +
-        "/agent/sessions/" + SessionID
-      streamingDetectIntentRequest.singleUtterance = true
-      streamingDetectIntentRequest.queryParams = getQueryParmasFor()
-      streamingDetectIntentRequest.queryInput = queryInput
-      streamingDetectIntentRequest.outputAudioConfig = getOutputAudioConfig()
-      writer.writeValue(streamingDetectIntentRequest)
+      streamingDetectIntentRequest.inputAudio = audioData as Data
+      self.writer.writeValue(streamingDetectIntentRequest)
     }
-
-    // send a request message containing the audio data
-    let streamingDetectIntentRequest = DFStreamingDetectIntentRequest()
-    streamingDetectIntentRequest.inputAudio = audioData as Data
-    writer.writeValue(streamingDetectIntentRequest)
   }
 
   func streamText(_ userInput: String, completion: @escaping StopwatchTextCompletionHandler) {
-    client = DFSessions(host:Host)
-    // send an initial request message to configure the service
-    let queryInput = DFQueryInput()
-    let inputTextConfig = DFTextInput()
-    inputTextConfig.text = userInput
-    inputTextConfig.languageCode = "en-US"
-    queryInput.text = inputTextConfig
-    let detectIntentRequest = DFDetectIntentRequest()
-    detectIntentRequest.session = "projects/" + ProjectName +
-      "/agent/sessions/" + SessionID
-    detectIntentRequest.queryInput = queryInput
-    detectIntentRequest.outputAudioConfig = getOutputAudioConfig()
-    detectIntentRequest.queryParams = getQueryParmasFor()
-    call = client.rpcToDetectIntent(with: detectIntentRequest, handler: { (response, error) in
-      completion(response, error as NSError?)
-    })
     // authenticate using an authorization token (obtained using OAuth)
-    call.requestHeaders.setObject(NSString(string:self.authorization()),
-                                  forKey:NSString(string:"Authorization"))
-    call.start()
+    authorization { (authT) in
+      self.client = DFSessions(host:Host)
+      // send an initial request message to configure the service
```

Comment:
This comment should be removed as this is only true fro streaming, not text

Reply:
Yeah, I will remove it.